Sources/Rendering/OpenGL/ShaderProgram/index.js

```diff
@@ -184,6 +184,26 @@ export function vtkShaderProgram(publicAPI, model) {
     return true;
   };
 
+  publicAPI.setUniform4f = (name, v) => {
+    const location = publicAPI.findUniform(name);
+    if (location === -1) {
+      model.error = `Could not set uniform ${name} . No such uniform.`;
+      return false;
+    }
+    model.context.uniform4f(location, v[0], v[1], v[2], v[3]);
+    return true;
+  };
+
+  publicAPI.setUniform4fv = (name, count, v) => {
+    const location = publicAPI.findUniform(name);
+    if (location === -1) {
+      model.error = `Could not set uniform ${name} . No such uniform.`;
+      return false;
+    }
+    model.context.uniform4fv(location, count, v);
```

Comment:
model.context.uniform4fv(location, v);

is correct and that is what it needs to be. Your error was probably that you were not passing in the right value. My guess is you were passing an array of arrays as opposed to just an array.